src/ripple/rpc/tests/KeyGeneration.test.cpp

```diff
@@ -164,10 +197,50 @@ class KeypairForSignature_test : public ripple::TestSuite
         testLegacySecret (secp256k1_strings.master_seed_hex);
     }
 
+    void testInvalidKeyType (char const* keyType)
+    {
+        testcase (keyType);
+
+        Json::Value params;
+        params[jss::key_type] = keyType;
+        params[jss::passphrase] = common::passphrase;
+
+        Json::Value error;
+        (void) keypairForSignature (params, error);
```

Comment:
It's unfortunate that we lack terms to distinguish between modifying a global variable and modifying an output parameter.

Anyway, I removed the casts added by this branch.